test/e2e/pd.go

```diff
@@ -214,7 +214,12 @@ func deletePD(pdName string) error {
 		zone := testContext.CloudConfig.Zone
 
 		// TODO: make this hit the compute API directly.
-		return exec.Command("gcloud", "compute", "--project="+testContext.CloudConfig.ProjectID, "disks", "delete", "--zone="+zone, pdName).Run()
+		cmd := exec.Command("gcloud", "compute", "--project="+testContext.CloudConfig.ProjectID, "disks", "delete", "--zone="+zone, pdName)
```

Comment:
I'm pretty sure that the error is that the PD is still in use (i.e. attached). In that case, retrying the deletion probably won't help (although retrying the detachment probably will).  As per @brendandburns, let's see what data this this unearths before deciding.  I'm fairly certain that what we'll want here in the end will be explicit confirmation that the detachment has completed (not just that the detach API call has succeeded), and also retries on the PD deletion.
As an aside, I'm not 100% convinced that this test should actually be made to fail if only the PD deletion fails.  That seems like a GCE rather than a K8S issue.  But I could be convinced either way.